include/wil/result_macros.h

```diff
@@ -1260,6 +1260,26 @@ WI_ODR_PRAGMA("WIL_FreeMemory", "0")
         } \
     } while ((void)0, 0)
 
+// Like 'FAIL_FAST_IF', but raises an assertion failure first for easier debugging
+#define FAIL_FAST_ASSERT(condition) \
+    do \
+    { \
+        if (!wil::verify_bool(condition)) \
+        { \
+            WI_ASSERT_FAIL(#condition); \
+            __RFF_FN(FailFast_Unexpected)(__RFF_INFO_ONLY(#condition)) \
+        } \
+    } while (0, 0)
+#define FAIL_FAST_ASSERT_MSG(condition, msg) \

```

Comment:
NOTE: No variadic args because it wouldn't be possible to use `msg` for `WI_ASSERT_FAIL`, and even if we could, it would be a bit of a pain to do so in a way that doesn't double evaluate the variadic arguments